ci-operator/step-registry/ovn/sdn-migration-rollback/ovn-sdn-migration-rollback-commands.sh

```diff
@@ -39,16 +39,30 @@ oc wait co network --for='condition=PROGRESSING=True' --timeout=120s
 timeout 300 oc rollout status ds/multus -n openshift-multus
 
 # Reboot all the nodes
-readarray -t POD_NODES <<< "$(oc get pod -n openshift-machine-config-operator -o wide| grep daemon|awk '{print $1" "$7}')"
-
-for i in "${POD_NODES[@]}"
-do
-    read -r POD NODE <<< "$i"
-    until oc rsh -n openshift-machine-config-operator "$POD" chroot /rootfs shutdown -r +1; do echo "cannot reboot node $NODE, retry"&&sleep 3; done
-done 
-
+# Get the list of nodes in the cluster
+master_nodes=$(oc get nodes -o jsonpath='{.items[?(@.metadata.labels.node-role\.kubernetes\.io/master=="")].metadata.name}')
+worker_nodes=$(oc get nodes -o jsonpath='{.items[?(@.metadata.labels.node-role\.kubernetes\.io/worker=="")].metadata.name}')
+daemonset_name="machine-config-daemon"
+delay=1
+# reboot all the master nodes in sequence
+for node in $master_nodes; do
+    pods_on_node=$(oc get pods -n openshift-machine-config-operator -o jsonpath='{.items[?(@.spec.nodeName=="'$node'")].metadata.name}' -l k8s-app=$daemonset_name)
+    if [[ -n $pods_on_node ]]; then
+        echo "reboot master node $node in ${delay}m"
+        until oc rsh -n openshift-machine-config-operator "$pods_on_node" chroot /rootfs shutdown -r +$delay; do echo "cannot reboot node $node, retry"&&sleep 3; done
+        delay=$((delay+3))
+    fi
+done
+# reboot all the worker nodes
+for node in $worker_nodes; do
+    pods_on_node=$(oc get pods -n openshift-machine-config-operator -o jsonpath='{.items[?(@.spec.nodeName=="'$node'")].metadata.name}' -l k8s-app=$daemonset_name)
+    if [[ -n $pods_on_node ]]; then
+        echo "reboot worker node $node in ${delay}m"
+        until oc rsh -n openshift-machine-config-operator "$pods_on_node" chroot /rootfs shutdown -r +$delay; do echo "cannot reboot node $node, retry"&&sleep 3; done
```

Comment:
@pliurh is delaying the workers reboot until after the masters have rebooted intentional? Or could they be rebooted at the same time?

Reply:
Yes, that is intentional. I found that rebooting the workers at the same time as the first master caused flakes. I haven't looked that in-depth, as it doesn't always happen. We might need to update the migration steps in the document to make it more stable.